src/mapObs.js

```diff
@@ -12,38 +12,35 @@ const throwError = (error) => {
 const MAPPERS_INFO = createSymbol('mappersInfo');
 const OBSERVABLES = createSymbol('observables');
 
-const createComponentFromMappers = (mappers, BaseComponent) => {
-  const factory = createEagerFactory(BaseComponent);
-
-  return class extends Component {
-    static [MAPPERS_INFO] = {mappers, BaseComponent};
+const createComponentFromMappers = (mappers, childFactory) =>
+  class extends Component {
+    static [MAPPERS_INFO] = {mappers, childFactory};
     static contextTypes = {[OBSERVABLES]: PropTypes.object};
     static childContextTypes = {[OBSERVABLES]: PropTypes.object};
 
     props$ = new BehaviorSubject(this.props);
-    state = {props: {}};
```

Comment:
```js
state = {};
```

It will generate an error if we don't do a setState before render (not guaranteed).